katdal/categorical.py

```diff
@@ -480,9 +517,10 @@ def concatenate_categorical(split_data, **kwargs):
     segments = np.cumsum([0] + [cat_data.events[-1] for cat_data in split_data])
     data = CategoricalData([], [])
     # Combine all unique values in the order they are found in datasets
-    split_values = [cat_data.unique_values for cat_data in split_data]
+    split_values = [cat_data._hashable_values for cat_data in split_data]
     inverse_splits = np.cumsum([0] + [len(vals) for vals in split_values])
-    data.unique_values, inverse = unique_in_order(np.concatenate(split_values), return_inverse=True)
+    data.unique_values, inverse = unique_in_order(sum(split_values, []),
```

Comment:
Unfortunately not. The ```unique_in_order``` function potentially runs through the sequence twice: once to find all the unique elements, and then again to find their associated indices if return_inverse is True.